app/Providers/LoadConfigServiceProvider.php

```diff
@@ -42,6 +42,25 @@ public function boot()
         $this->app->config->set('app.url', $appDomain ?: $this->app->config->get('app.url'));
         $this->app->config->set('app.locale', $appLocale ?: $this->app->config->get('app.locale'));
 
+        // Setup Cors.
+        $allowedOrigins = [
+            Setting::get('app_domain'),
+        ];
+
+        // Add our allowed domains too.
+        if ($allowedDomains = Setting::get('allowed_domains')) {
+            $domains = explode(',', $allowedDomains);
+            foreach ($domains as $domain) {
+                $allowedOrigins[] = $domain;
+            }
+        } else {
+            $allowedOrigins[] = getenv('APP_URL');
+        }
+
+        $this->app->config->set('cors.defaults.allowedOrigins', $allowedOrigins);
```

Comment:
What do you think if we read the config first and then merge it with the DB values? So you can explicitly set them on the config file or on the DB.